app/com/linkedin/drelephant/tuning/AutoTuningAPIHelper.java

```diff
@@ -172,8 +174,39 @@ private void applyPenalty(String jobExecId) {
     jobExecution.executionTime = 0D;
     jobExecution.inputSizeInBytes = 1D;
     jobExecution.save();
+
+    //In case this is best param set, we need to find a new best parameter in case of HBT
+    if (isBestParamSet && tuningJobDefinition.tuningAlgorithm.optimizationAlgo == OptimizationAlgo.HBT) {
+      findNewBestParam(tuningJobDefinition);
+    }
   }
 
+  /**
+   * This method find new best parameter in case current best resulted in failure
+   * @param tuningJobDefinition
+   */
+  private void findNewBestParam(TuningJobDefinition tuningJobDefinition) {
```

Comment:
Should we also compare on resource usage if fitness is same , the way you are doing in FitnessManagerHBT

Reply:
Are there any possibility of race condition with the FitneMangerHBT ,findBestParameter ?

Reply:
Can we have unit test case for this function?

Reply:
- Used resource usage as well for finding best parameters. 
- Using only FitnessComputed param set now, so no race condition. 
- Have done the unit testing. As this need to be fixed on priority will write the unit test cases in next PR.

Reply:
LGTM